lib/rules/store-connect-deps.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const traverse = require('eslint-traverse');
+
+const notStoreDep = 'Expected a store';
+const missingStoreDeps = 'Missing stores {{stores}} that are used in callback';
+const extraStoreDeps = 'Unused stores {{stores}} in stores array';
+
+module.exports = {
+  meta: {
+    messages: {
+      notStoreDep,
+      missingStoreDeps,
+      extraStoreDeps,
+    },
+    docs: {
+      description: 'Ensure connect store dependency arrays match what is used in the function',
+      category: 'Code Safety',
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        properties: {},
+        additionalProperties: false,
+      },
+    ],
+  },
+
+  create: function (context) {
+    return {
+      CallExpression(node, parent) {
+        let isConnectStores =
+          node.callee.type === 'MemberExpression' &&
+          node.callee.object.type === 'Identifier' &&
+          node.callee.object.name === 'Flux' &&
+          node.callee.property.type === 'Identifier' &&
+          node.callee.property.name === 'connectStores';
+        let isUseStateFromStoresHook =
+          node.callee.type === 'Identifier' &&
+          (node.callee.name === 'useStateFromStores' ||
+            node.callee.name === 'useStateFromStoresObject' ||
+            node.callee.name === 'useStateFromStoresArray');
+        if (!isConnectStores && !isUseStateFromStoresHook) {
+          return;
+        }
+
+        let args = node.arguments;
+
+        let storeDepsArg = args[0];
+        if (storeDepsArg.type !== 'ArrayExpression') {
+          return;
+        }
+
+        // Get the array of stores
+        let inputDeps = [];
+        storeDepsArg.elements.forEach((element) => {
+          if (element.type === 'Identifier' && element.name.endsWith('Store')) {
+            inputDeps.push(element.name);
+          } else {
+            context.report({
+              node: element,
+              message: notStoreDep,
+            });
+          }
+        });
+
+        // Get all stores that are used in the function body
+        let actualDeps = [];
+
+        traverse(context, args[1], (path) => {
+          let node = path.node;
+          if (node.type !== 'MemberExpression' || node.object.type !== 'Identifier') {
```

Comment:
Would it be better to just look for Identifiers with the name ending in `"Store"`? `traverse` can walk through the MemberExpression.

Reply:
Just thinking about cases like: `selectDataFrom(MyStore)`

The safest way to make sure that you're referencing a store is to look up the identifier's binding in scope.

Reply:
that's a good point, that will probably work better. the counterpoint is if someone assigns a variable like dataFromStore, but it'd be pretty easy to recognize that issue